### PR TITLE
Simplify stream concatenation

### PR DESCRIPTION
**Pull request by idegtiarenko**
This change simplifies the Stream concatenation. It replaces:
* reduce/concat for 2 arguments usage with a simple concat
* reduce/concat for multiple streams with a flatMap.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java [hunk 1]
```diff
@@ -291,15 +291,12 @@ private void checkObjectDepthLimit(long limit) {
...
-        Stream.of(objectMappers.values().stream(), fieldMappers.values().stream())
-            .reduce(Stream::concat)
-            .orElseGet(Stream::empty)
-            .forEach(mapper -> {
-                String name = mapper.simpleName();
-                if (name.length() > limit) {
-                    throw new IllegalArgumentException("Field name [" + name + "] is longer than the limit of [" + limit + "] characters");
-                }
-            });
+        Stream.concat(objectMappers.values().stream(), fieldMappers.values().stream()).forEach(mapper -> {
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java:294]: Could we just solve this with a normal loop (or two)? The stream stuff is just needless overhead here I think.

File: server/src/main/java/org/elasticsearch/common/util/iterable/Iterables.java [hunk 2]
```diff
@@ -35,11 +35,7 @@
 
         @Override
         public Iterator<T> iterator() {
-            return Stream.of(inputs)
-                .map(it -> StreamSupport.stream(it.spliterator(), false))
-                .reduce(Stream::concat)
-                .orElseGet(Stream::empty)
-                .iterator();
+            return Stream.of(inputs).flatMap(it -> StreamSupport.stream(it.spliterator(), false)).iterator();
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/common/util/iterable/Iterables.java:38]: I looked into the uses of this method (and concatenating iterables in general). It appears there's literally only 3 spots and we already have concatenated iterators. Maybe we could just do away with the whole concept of concatenated iterable outright and replace the 3 use cases with either concatenating iterators or just doing some other simple thing? :)

**Review by original-brownbear — COMMENTED** [review]
Looks better than what we have, but I left some comments wondering if we can go for even simpler here. For the mapping lookup checks in particular, simpler and faster would make the many-shards benchmarks happier :)

File: server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java (hunk 1 above)
Comment (idegtiarenko) [reply server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java:294]: Yeah, good point

File: server/src/main/java/org/elasticsearch/common/util/iterable/Iterables.java (hunk 2 above)
Comment (idegtiarenko) [reply server/src/main/java/org/elasticsearch/common/util/iterable/Iterables.java:38]: Correct. I would like to merge this as is (simplify concatenation in various places) and then open a followup pr to remove/replace operations from `org.elasticsearch.common.util.iterable.Iterables` that are no longer needed.

**Review by original-brownbear — APPROVED** [review]

**Pull request merged**